xray/grpc.go

```diff
@@ -0,0 +1,174 @@
+package xray
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"net/http"
+	"net/http/httptrace"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/golang/protobuf/proto"
+	"go.uber.org/multierr"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/aws/aws-xray-sdk-go/header"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+// UnaryClientInterceptor provides gRPC unary client interceptor.
+func UnaryClientInterceptor(host string) grpc.UnaryClientInterceptor {
+	return func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+		return Capture(ctx, host, func(ctx context.Context) error {
+			seg := GetSegment(ctx)
+			if seg == nil {
+				return errors.New("failed to record gRPC transaction: segment cannot be found")
+			}
+
+			ct, e := NewClientTrace(ctx)
+			if e != nil {
+				return e
+			}
+			md := metadata.Pairs(TraceIDHeaderKey, seg.DownstreamHeader().String())
+			ctx2 := metadata.NewOutgoingContext(httptrace.WithClientTrace(ctx, ct.httpTrace), md)
+
+			seg.Lock()
+			seg.Namespace = "remote"
+			seg.GetHTTP().GetRequest().URL = "grpc://" + host + method
+			seg.GetHTTP().GetRequest().Method = method
```

Comment:
This needs to be an HTTP method, not gRPC method. It's always `POST`.

Reply:
Fixed